src/Raven.Server/RavenServer.cs

```diff
@@ -153,7 +155,10 @@ public RavenServer(RavenConfiguration configuration)
             _tcpContextPool = new JsonContextPool(Configuration.Memory.MaxContextSizeToKeep);
 
             // doing this before the schema upgrade to allow to downgrade in case we cannot start the server
-            BeforeSchemaUpgrade = x => VerifyLicense(x, ServerStore);
+            BeforeSchemaUpgrade = VerifyLicense;
+
+            if (Configuration.Licensing.EnforceLicense)
+                AfterDatabaseCreation = VerifyLicense;
```

Comment:
why this is needed? can you describe me the flow?

Reply:
We have two scenarios for running an Embedded server with `EnforceLicense == true`:

1. Starting a fresh server without a System database.
2. Starting a previously used server with an existing System database.

In the first case, we perform the license verification at the final stage of `LoadExistingDatabase` before starting `UpgradeSchemaIfRequired`.

In the second case, the verification is done immediately after creating the System database.

I could rename the `BeforeSchemaUpgrade` delegate to something like `VerifyLicenseAction` and use it in both cases...

Reply:
Leave it, it is fine